### PR TITLE
fix: [CDS-34081]: Templates page support added

### PR DESCRIPTION
**Pull request by samarth-sharma31**
##### Summary:

Templates Input page added

##### Jira Links:

https://harness.atlassian.net/browse/CDS-34081

##### Screenshots:

Screenshot 2022-03-16 at 2 11 20 AM
Screenshot 2022-03-16 at 2 11 28 AM
Screenshot 2022-03-16 at 2 11 43 AM
Screenshot 2022-03-16 at 2 11 57 AM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by samarth-sharma31** [comment]
trigger cypress

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/72-templates-library/components/TemplateStudio/RightDrawer/RightDrawer.tsx [hunk 1]
```diff
@@ -37,6 +59,20 @@ export const RightDrawer: React.FC = (): JSX.Element => {
   const { CDNG_ENABLED, CING_ENABLED } = useFeatureFlags()
   const [stepPaletteModuleInfos, setStepPaletteModuleInfos] = React.useState<StepPalleteModuleInfo[]>([])
 
+  const selectedTemplateDetails = useMemo(
+    () => ({
```
Comment (yc147) [inline src/modules/72-templates-library/components/TemplateStudio/RightDrawer/RightDrawer.tsx:63]: Replace line 64-69 with `...template`

**Comment by samarth-sharma31** [comment]
retrigger build

**Comment by samarth-sharma31** [comment]
retrigger eslint

**Comment by samarth-sharma31** [comment]
trigger cypress

**Comment by samarth-sharma31** [comment]
retrigger titlecheck

**Comment by samarth-sharma31** [comment]
retrigger titlecheck

**Comment by samarth-sharma31** [comment]
retrigger standards

**Review by yc147 — APPROVED** [review]

**Comment by samarth-sharma31** [comment]
retrigger typecheck

**Comment by samarth-sharma31** [comment]
retrigger jest

**Comment by samarth-sharma31** [comment]
retrigger build

**Comment by samarth-sharma31** [comment]
retrigger standards

**Comment by samarth-sharma31** [comment]
retrigger coverage

**Comment by samarth-sharma31** [comment]
retrigger prettier

**Comment by samarth-sharma31** [comment]
trigger cypress

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by yc147 — APPROVED** [review]

**Pull request merged**